python/lsst/pex/config/configDictField.py

```diff
@@ -76,7 +76,7 @@ def __setitem__(self, k, x, at=None, label="setitem", setHistory=True):
             if x == dtype:
                 self._dict[k] = dtype(__name=name, __at=at, __label=label)
             else:
-                self._dict[k] = dtype(__name=name, __at=at, __label=label, **x._storage)
+                self._dict[k] = x.__class__(__name=name, __at=at, __label=label, **x._storage)
```

Comment:
Why did you use `__class__` here instead of `type`?

Reply:
I can't remember having a good reason so I changed it to `type`.